spec/game_spec.rb

```diff
@@ -1,53 +1,59 @@
-require './lib/game'
-require './lib/player'
+require './lib/game.rb'
+require './lib/player.rb'
 
-RSpec.describe 'Game methods' do
-  let(:player1) { Player.new('Lamia', 'x') }
-  let(:player2) { Player.new('Mike', 'o') }
+RSpec.describe 'testing the game methods' do
+  let(:player1) { Player.new('Grace', 'x') }
+  let(:player2) { Player.new('Olamide', 'o') }
   let(:game) { Game.new(player1, player2) }
 
-  it 'should return an array of x and o' do
-    expect(Game.choose_signs.sort).to eql(%w[x o].sort)
+  it 'it chooses sign for players' do
+    arr = Game.choose_signs
+    expect(arr.sort).to eq(%w[x o].sort)
   end
 
-  context 'Validate the moves' do
-    it 'should check for the numbers between 1 to 9' do
-      expect(game.validate_move(3)).to eql(3)
+  context 'validate the players move' do
+    it 'should check the moves between 1 to 9' do
+      expect(game.validate_move(6)).to eq(6)
     end
 
-    it 'should return -2 for string inputs' do
-      # p game.validate_move('abc')
-      expect(game.validate_move('abc')).to eql(-2)
+    it 'should check if number is invalid' do
+      expect(game.validate_move('p')).to eq(-2)
     end
+  end
 
-    it 'will check for unique numbers' do
-      game.choosed_nums = [5]
-      expect(game.validate_move(5)).to eql(-1)
-    end
+  it 'validates player name if greater than 3' do
+    expect(Game.validate_name('Grace')).to be(true)
   end
 
-  it 'should return true if input length is grater than 2' do
-    expect(Game.validate_name('Lamia')).to be true
+  it 'validates player name if not less than 3' do
+    expect(Game.validate_name('ab')).to be(false)
   end
 
-  it 'should return array of players' do
-    expect(game.getcurrent_player).to eql([player1, player2])
+  it 'get the current player' do
+    expect(player1.name).to eq('Grace')
+    expect(player1.sign).to eq('x')
   end
 
-  context 'check winner?_method' do
-    it 'a draw' do
-      game.choosed_nums = [1, 2, 3, 4, 5, 6, 7, 8, 9]
-      expect(game.winner?).to eql('Draw')
+  context 'Check Winner' do
+    it 'Horizontal' do
+      game.board.update_board(1, 'x')
+      game.board.update_board(2, 'x')
+      game.board.update_board(3, 'x')
+      expect(game.winner?).to eq('Grace')
     end
 
-    it 'winner is player 1' do
-      game.board.squares = [
-        [{ '1': nil }, { '2': 'x' }, { '3': nil }],
-        [{ '4': nil }, { '5': 'x' }, { '6': nil }],
-        [{ '7': nil }, { '8': 'x' }, { '9': nil }]
-      ]
+    it 'Vertical' do
+      game.board.update_board(1, 'o')
+      game.board.update_board(4, 'o')
+      game.board.update_board(7, 'o')
+      expect(game.winner?).to eq('Olamide')
+    end
 
-      expect(game.winner?). to eql('Lamia')
+    it 'Diagonal' do
+      game.board.update_board(1, 'o')
+      game.board.update_board(5, 'o')
+      game.board.update_board(9, 'o')
+      expect(game.winner?).to eq('Olamide')
```

Comment:
- [ ] Please add a spec that checks for negative scenarios / incorrect usage of the method.